### PR TITLE
optionally record user canceling tasks and pipelines

### PR DESCRIPTION
**Pull request by anotherchrisberry**
@ajordens how much are you in love with this?

### CODE REVIEW

File: orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/persistence/ExecutionRepositoryTck.groovy [hunk 1]
```diff
@@ -353,6 +353,30 @@ abstract class ExecutionRepositoryTck<T extends ExecutionRepository> extends Spe
       status == RUNNING
     }
   }
+
+  def "cancelling a running execution with a user adds a 'canceledBy' field"() {
+    given:
+    def execution = new Pipeline(buildTime: 0)
+    def user = "chrisb@netflix.com"
```
Comment (ajordens) [inline orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/persistence/ExecutionRepositoryTck.groovy]: You probably don't want to include your actual email address in here.

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy [hunk 2]
```diff
@@ -93,8 +93,8 @@ class TaskController {
 
   @RequestMapping(value = "/tasks/{id}/cancel", method = RequestMethod.PUT)
   @ResponseStatus(HttpStatus.ACCEPTED)
-  void cancelTask(@PathVariable String id) {
-    executionRepository.cancel(id)
+  void cancelTask(@PathVariable String id, @RequestParam(value = "user", required = false) String user) {
```
Comment (ajordens) [inline orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy]: We don't/shouldn't be passing 'user' in as a request parameter.

I reckon you could do `AuthenticatedRequest.getSpinnakerUser().orElse("anonymous")`.

That's what we do in StageBuilder to track who restarted a pipeline (also do it from TaskController for tracking lastModifiedBy).

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy [hunk 3]
```diff
@@ -109,8 +109,8 @@ class TaskController {
 
   @RequestMapping(value = "/pipelines/{id}/cancel", method = RequestMethod.PUT)
   @ResponseStatus(HttpStatus.ACCEPTED)
-  void cancel(@PathVariable String id) {
-    executionRepository.cancel(id)
+  void cancel(@PathVariable String id, @RequestParam(value = "user", required = false) String user) {
```
Comment (ajordens) [inline orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy]: Ditto above.

Test it out but it's better than forged RequestParam's.

**Comment by anotherchrisberry** [comment]
@ajordens updated to use `AuthenticatedRequest`; also updated the OperationsController to pick up the user the same way from the `/orchestrate` endpoint

**Comment by ajordens** [comment]
LGTM ... needs a rebase.

**Pull request merged**